alicloud/data_source_alicloud_nas_accessgroups_test.go

```diff
@@ -0,0 +1,74 @@
+package alicloud
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAlicloud_AccessGroup_DataSource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAlicloudAccessGroupsDataSource,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudDataSourceID("data.alicloud_nas_accessgroups.ag"),
+					resource.TestCheckResourceAttrSet("data.alicloud_nas_accessgroups.ag", "accessgroups.0.rule_count"),
+					resource.TestCheckResourceAttrSet("data.alicloud_nas_accessgroups.ag", "accessgroups.0.accessgroup_type"),
+					resource.TestCheckResourceAttrSet("data.alicloud_nas_accessgroups.ag", "accessgroups.0.description"),
+					resource.TestCheckResourceAttrSet("data.alicloud_nas_accessgroups.ag", "accessgroups.0.accessgroup_name"),
+					resource.TestCheckResourceAttrSet("data.alicloud_nas_accessgroups.ag", "accessgroups.0.mounttarget_count"),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
Please add more testcases: filter groups by name_regex, filter groups by type, filter groups by description, and filter groups by all of parameters.